native/dlasq4.go

```diff
@@ -0,0 +1,228 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "math"
+
+// Dlasq4 computes an approximation to the smallest eigenvalue using values of d
+// from the previous transform.
+// i0, n0, and n0in are zero-indexed.
+func (impl Implementation) Dlasq4(i0, n0 int, z []float64, pp int, n0in int, dmin, dmin1, dmin2, dn, dn1, dn2, tau float64, ttype int, g float64) (tauOut float64, ttypeOut int, gOut float64) {
+	const (
+		cnst1 = 0.563
+		cnst2 = 1.01
+		cnst3 = 1.05
+
+		cnstthird = 0.333 // TODO(btracey): Fix?
+	)
+
+	if dmin <= 0 {
+		tau = -dmin
+		ttype = -1
+		return tau, ttype, g
+	}
+	nn := 4*(n0+1) + pp - 1 //-1 for zero indexing
+	s := math.NaN()         // Make sure one of the paths below is taken
```

Comment:
You are not making sure one is taken, you are marking when one is not.

"Poison s so that failure to take a path below is obvious."